crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -126,6 +126,14 @@ pub struct MqttMessageBox {
     peer_senders: Vec<(TopicFilter, LoggingSender<MqttMessage>)>,
 }
 
+pub struct FromPeers<'a> {
+    input_receiver: &'a mut LoggingReceiver<MqttMessage>,
+}
+
+pub struct ToPeers<'a> {
+    peer_senders: &'a mut Vec<(TopicFilter, LoggingSender<MqttMessage>)>,
```

Comment:
We could have just cloned that `Vec` to avoid these lifetime parameters, but its' okay, as it is consistent with its counterpart.

Reply:
I don't think the lifetimes are that complicated, and I'll modify the `run(self)` PR to remove these once this is merged as they won't be necessary after that.